master/buildbot/process/buildrequest.py

```diff
@@ -267,7 +267,18 @@ def canBeCollapsed(master, br1, br2):
         if br1['buildsetid'] == br2['buildsetid']:
             defer.returnValue(True)
 
-        # get the buidlsets for each buildrequest
+        # Don't collapse e.g. GitLab merge requests with normal builds
+        bsp1 = yield master.db.buildsets.getBuildsetProperties(br1['buildsetid'])
```

Comment:
I would prefer that you use the data API to access this (as people may copy this code in their own master config as an example)

Reply:
Tried that just now, but:
```
-        bsp1 = yield master.db.buildsets.getBuildsetProperties(br1['buildsetid'])
-        bsp2 = yield master.db.buildsets.getBuildsetProperties(br2['buildsetid'])
+        bsp1 = yield master.data.get(('buildsets', br1['buildsetid'], 'properties'))
+        bsp2 = yield master.data.get(('buildsets', br1['buildsetid'], 'properties'))
```
fails with
```
twisted.trial.unittest.FailTest: Lists differ: [20] != [17, 18, 19, 20]
```
in test_collapseRequests_collapse_default_with_a_merge_request

Reply:
@dankegel for `bsp2` you probably meant to use `br2['buildsetid']` rather than `br1['buildsetid']`?

Reply:
Quite.  Can I blame "coding immediately after waking up for a midnight snack"?  :-) Fixed.